scripts/release_prep.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+set -x
+
+# pre-requisites: install github CLI
+# - github documentation: https://github.com/cli/cli#installation
+# - github is remote 'origin'
+# - PRs use test section LAST with heading "## Tests"
+# - ALL build and release PRs start with "build: "
+#
+# To run, from root of project:
+# bash scripts/release_prep.sh
+
+has_local_changes=$(git status --porcelain --untracked-files=no --ignored=no)
+if [[ ${has_local_changes} ]]; then
+  set +x
+  echo ==========
+  echo "ABORT: You have local modifications. Please stash or commit changes and run again."
+  echo ==========
+  exit 1
+fi
+
+git fetch --all --tags
+git reset --hard
+git pull
+git checkout develop
+git reset --hard origin/develop
+
+short_hash=$(git rev-parse --short HEAD)
+current_version=v$(node -p "require('./package.json').version")
+temp_release_branch=temp_${short_hash}
+
+git checkout -b ${temp_release_branch}
+
+release_version=$(npm --no-git-tag-version version minor | grep -E '^v\d')
+release_branch=release_${release_version}
+may_force_push=
+
+if [[ "$1" == "--recut" ]]; then
+  git tag -d ${release_version}
+  git push --delete origin ${release_version}
+  git branch -D ${release_branch}
+  may_force_push=-f
+fi
+
+git commit -a -n -m "chore: bump version to ${release_version}"
+git tag ${release_version}
+git checkout -b ${release_branch}
+git branch -D ${temp_release_branch}
+
+git push origin ${may_force_push} HEAD:${release_branch}
+git push -f origin HEAD:staging
+git push origin ${release_version}
+
+# extract changelog to inject into the PR
+pr_body_file=.pr_body_${release_version}
+pr_body_file_groupped=.pr_body_${release_version}_groupped
+
+awk "/^#### \[${release_version}\]/{flag=1;next}/####/{flag=0}flag" CHANGELOG.md | sed -E '/^([^-]|[[:space:]]*$)/d' > ${pr_body_file}
```

Comment:
can we add a comment here? not entirely sure i understand what this is doing HAHA

Reply:
Can! It's loading the entire section of changes from the current release. If the current release is v0.71.0, it basically grabbing everything that's in between the headings
```
#### v0.71.0
- fix:...
- chore:...
...

#### v0.70.0
```

and it discards lines which only contain whitespaces and empty bullet points.

Reply:
added a comment: https://github.com/isomerpages/isomercms-backend/pull/1209/commits/9653429ff980cfafcdb72246de5f31c3e2ce81c7